zaza/openstack/charm_tests/ceph/fs/tests.py

```diff
@@ -16,42 +16,101 @@
 
 import logging
 import asyncio
+import subprocess
 from tenacity import retry, Retrying, stop_after_attempt, wait_exponential
+import unittest
+import zaza
 import zaza.model as model
-import zaza.openstack.charm_tests.neutron.tests as neutron_tests
-import zaza.openstack.charm_tests.nova.utils as nova_utils
 import zaza.openstack.charm_tests.test_utils as test_utils
-import zaza.openstack.configure.guest as guest
 
-from zaza.openstack.utilities import (
-    openstack as openstack_utils,
-)
 
-
-class CephFSTests(test_utils.OpenStackBaseTest):
+class CephFSTests(unittest.TestCase):
     """Encapsulate CephFS tests."""
 
-    RESOURCE_PREFIX = 'zaza-cephfstests'
-    INSTANCE_USERDATA = """#cloud-config
-packages:
-- ceph-fuse
-- python
-mounts:
-  - [ 'none', '/mnt/cephfs', 'fuse.ceph', 'ceph.id=admin,ceph.conf=/etc/ceph/ceph.conf,_netdev,defaults', '0', '0' ]
-write_files:
--   content: |
-{}
-    path: /etc/ceph/ceph.conf
--   content: |
-{}
-    path: /etc/ceph/ceph.client.admin.keyring
-""" # noqa
+    mounts_share = False
+    mount_dir = '/mnt/cephfs'
+
+    def tearDown(self):
+        """Cleanup after running tests."""
+        if self.mounts_share:
+            try:
+                zaza.utilities.generic.run_via_ssh(
+                    unit_name='ubuntu/0',
+                    cmd='sudo fusermount -u {0} && sudo rmdir {0}'.format(
+                        self.mount_dir))
+                zaza.utilities.generic.run_via_ssh(
+                    unit_name='ubuntu/1',
+                    cmd='sudo fusermount -u {0} && sudo rmdir {0}'.format(
+                        self.mount_dir))
+            except subprocess.CalledProcessError:
+                logging.warning("Failed to cleanup mounts")
```

Comment:
This won't even attempt to cleanup the mounts on the second unit if the first unit fails (which I think we should do). Additionally, we may want to augment the logging to indicate which unit?